pkg/statistics/handle/storage/read.go

```diff
@@ -652,6 +652,8 @@ func loadNeededColumnHistograms(sctx sessionctx.Context, statsHandle statstypes.
 
 	_, loadNeeded, analyzed := statsTbl.ColumnIsLoadNeeded(col.ID, true)
 	if !loadNeeded || !analyzed {
+		// If this column is not analyzed yet and we don't have it in memory.
+		// We create a fake one for the pseudo estimation.
```

Comment:
```suggestion
		// If this column is not analyzed yet and we don't have it in memory.
		// We create a fake one for the pseudo estimation.
		// Otherwise, it will trigger the sync/async load again, even if the column has not been analyzed.
```